tracing-subscriber/tests/subscriber_filters/option.rs

```diff
@@ -38,3 +38,23 @@ fn option_none() {
 
     handle.assert_finished();
 }
+
+#[test]
+fn option_mixed() {
+    let (subscribe, handle) = subscriber::mock()
+        .event(expect::event())
+        .only()
+        .run_with_handle();
+    let subscribe = subscribe
+        .with_filter(filter::dynamic_filter_fn(|meta, _ctx| {
+            meta.target() == "interesting"
+        }))
+        .with_filter(None::<filter::DynFilterFn<_>>);
```

Comment:
this is worth having, but the scenario i _really_ wanted to test was not a single subscriber with two different filters, but two different subscribers with separate filters, where one subscriber's filter was a `None`. in particular, we want to assert that the `None` filter does not inadvertantly enable anything that the other subscriber's filter would enable.

we also really want to test that the correct max level hint is used, since that's something we've had bugs with in the past.